### PR TITLE
fix(dashboard-detail-info): add updateWidgetValidation action

### PR DESCRIPTION
**Pull request by yuda110**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [x] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [x] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] `Error / Warning / Lint / Type`

### Description
SSIA!

### CODE REVIEW

File: src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts [hunk 1]
```diff
@@ -105,15 +103,9 @@ export const useDashboardDetailInfoStore = defineStore('dashboard-detail-info',
...
-        widgetInheritVariablesValidMap: computed(() => {
-            const result = {};
-            // originState.dashboardWidgetInfoList.forEach(() => {
-            //     result[d.widget_key] = isWidgetValid(d, state.variables_schema);
-            // });
-            return result;
-        }),
-    });
+        isWidgetLayoutValid: computed(() => Object.values(validationState.widgetInheritVariablesValidMap).every((d) => d === true)),
+        widgetInheritVariablesValidMap: {},
```
Comment (yuda110) [inline src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts:107]: description: ex.
```typescript
{
  'widget-xxxx': true,
  'widget-trend-xxxx': false,
}
```

File: src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts [hunk 2]
```diff
@@ -105,15 +103,9 @@ export const useDashboardDetailInfoStore = defineStore('dashboard-detail-info',
...
-        isWidgetLayoutValid: computed(() => ({})), // is all widgets valid
-        widgetInheritVariablesValidMap: computed(() => {
-            const result = {};
-            // originState.dashboardWidgetInfoList.forEach(() => {
-            //     result[d.widget_key] = isWidgetValid(d, state.variables_schema);
-            // });
-            return result;
-        }),
-    });
+        isWidgetLayoutValid: computed(() => Object.values(validationState.widgetValidMap).every((d) => d === true)),
```
Comment (WANZARGEN) [inline src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts:106]: praise: very clear! 👍

File: src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts [hunk 3]
```diff
@@ -105,15 +103,9 @@ export const useDashboardDetailInfoStore = defineStore('dashboard-detail-info',
...
+        widgetValidMap: {},
+    }) as UnwrapRef<ValidationState>;
```
Comment (WANZARGEN) [inline src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts:108]: issue: I think widgetValidMap must be reset on resetDashboardData() or setDashboardInfo().

**Review by WANZARGEN — COMMENTED** [review]
I left some comments!!

File: src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts [hunk 4]
```diff
@@ -58,8 +56,8 @@ interface DashboardDetailInfoStoreState {
 }
 interface ValidationState {
     isNameValid?: boolean;
-    isWidgetLayoutValid: ComputedRef<Record<string, boolean>>;
-    widgetInheritVariablesValidMap: ComputedRef<WidgetInheritVariablesValidMap>;
+    isWidgetLayoutValid: ComputedRef<boolean>;
+    widgetInheritVariablesValidMap: WidgetInheritVariablesValidMap;
```
Comment (sulmoJ) [inline src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts]: Question: Does the validation of the form not include? If it's included, I think we need to modify the variable name..!

**Review by sulmoJ — APPROVED** [review]
LGTM!!
I left a question!!

**Pull request merged**